libraries/Microsoft.Bot.Builder.AI.Orchestrator/OrchestratorRecognizer.cs

```diff
@@ -0,0 +1,346 @@
+        /// </value>
+        [JsonProperty("detectAmbiguousIntents")]
+        public BoolExpression DetectAmbiguousIntents { get; set; } = false;
+
+        /// <summary>
+        /// Return recognition results.
+        /// </summary>
+        /// <param name="dc">Context object containing information for a single turn of conversation with a user.</param>
+        /// <param name="activity">The incoming activity received from the user. The Text property value is used as the query text for QnA Maker.</param>
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects or threads to receive notice of cancellation.</param>
+        /// <param name="telemetryProperties">Additional properties to be logged to telemetry with the LuisResult event.</param>
+        /// <param name="telemetryMetrics">Additional metrics to be logged to telemetry with the LuisResult event.</param>
+        /// <returns>A <see cref="RecognizerResult"/> containing the QnA Maker result.</returns>
+        public override async Task<RecognizerResult> RecognizeAsync(DialogContext dc, Schema.Activity activity, CancellationToken cancellationToken, Dictionary<string, string> telemetryProperties = null, Dictionary<string, double> telemetryMetrics = null)
+        {
+            var text = activity.Text ?? string.Empty;
+            var detectAmbiguity = DetectAmbiguousIntents.GetValue(dc.State);
+
+            _modelPath = ModelPath.GetValue(dc.State);
+            _snapshotPath = SnapshotPath.GetValue(dc.State);
+
+            InitializeModel();
+
+            var recognizerResult = new RecognizerResult()
+            {
+                Text = text,
+                Intents = new Dictionary<string, IntentScore>(),
+            };
+
+            if (string.IsNullOrWhiteSpace(text))
+            {
+                // nothing to recognize, return empty recognizerResult
+                return recognizerResult;
+            }
+
+            // Score with orchestrator
+            Stopwatch sw = new Stopwatch();
+            sw.Start();
+            var results = this._resolver.Score(text);
+            sw.Stop();
+            Console.WriteLine($"Orchestrator recognize : {sw.ElapsedMilliseconds}");
+
+            // Add full recognition result as a 'result' property
+            recognizerResult.Properties.Add(ResultProperty, results);
+
+            if (results.Any())
+            {
+                var topScore = results[0].Score;
+
+                // if top scoring intent is less than threshold, return None
+                if (topScore < UnknownIntentFilterScore)
+                {
+                    recognizerResult.Intents.Add(NoneIntent, new IntentScore() { Score = 1.0 });
+                }
+                else
+                {
+                    // add top score
+                    AddResultAsIntent(recognizerResult, results[0]);
```

Comment:
i would just inline this code - the helper doesn't really help readability

Reply:
fixed.